Sources/FoundationEssentials/AttributedString/AttributedString+IndexValidity.swift

```diff
@@ -0,0 +1,93 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(Synchronization)
```

Comment:
In other places we have `#if canImport(Synchronization) && FOUNDATION_FRAMEWORK`. Is that wrong?

Reply:
Previously we needed the `&& FOUNDATION_FRAMEWORK` because non-`FOUNDATION_FRAMEWORK` builds ran on a very old version of macOS in CI where we back deployed back to a version that didn't have `Synchronization` (even though the `canImport(Synchronization)` check passes because it was in the SDK). Now that we've bumped the deployment target of the package, we can safely use the `Synchronization` module on the macOS version CI is back deployed to, so no need to guard with `FOUNDATION_FRAMEWORK` (but there are still some edge case situations where the module doesn't exist at all that we need to guard for with the `canImport` check)